### PR TITLE
Adding `create_treatment_history` Function

### PR DESCRIPTION
**Pull request by Jay-sanjay**
Added Treatment history functinon

### CODE REVIEW

File: src/preprocessing.jl [hunk 1]
```diff
@@ -26,4 +26,63 @@ function Dummy(
...
+
+    # Only keep event cohorts starting (startDate) or ending (endDate) after target cohort start date
+    if includeTreatments == "startDate"
+        #println("Applying startDate filtering")
+        #println("periodPriorToIndex: ", periodPriorToIndex)
+        #println("cohort_start_date_1: ", current_cohorts.cohort_start_date_1)
+        #("cohort_start_date: ", current_cohorts.cohort_start_date)
+        #println("cohort_end_date_1: ", current_cohorts.cohort_end_date_1)
+
+        current_cohorts = current_cohorts[
```
Comment (TheCedarPrince) [inline src/preprocessing.jl:265]: Even though this is gnarly from the the R package, comparing between the R code and your code, this looks right to me.

File: src/preprocessing.jl [hunk 2]
```diff
@@ -26,4 +26,63 @@ function Dummy(
...
+            (current_cohorts.cohort_start_date_1 .- (periodPriorToIndex) .<= current_cohorts.cohort_start_date) .& 
+            (current_cohorts.cohort_start_date .< current_cohorts.cohort_end_date_1), :]
+    elseif includeTreatments == "endDate"
```
Comment (TheCedarPrince) [inline src/preprocessing.jl:269]: This elseif also looks correct.

File: src/preprocessing.jl [hunk 3]
```diff
@@ -26,4 +26,63 @@ function Dummy(
...
+        #println("cohort_start_date_1: ", current_cohorts.cohort_start_date_1)
+        #println("cohort_end_date: ", current_cohorts.cohort_end_date)
+        #println("cohort_end_date_1: ", current_cohorts.cohort_end_date_1)
+
+        current_cohorts = current_cohorts[
+            (current_cohorts.cohort_start_date_1 .- (periodPriorToIndex) .<= current_cohorts.cohort_end_date) .& 
+            (current_cohorts.cohort_start_date .< current_cohorts.cohort_end_date_1), :]
+        current_cohorts.cohort_start_date = max.(
+            current_cohorts.cohort_start_date_1 .- (periodPriorToIndex), current_cohorts.cohort_start_date)
+    else
```
Comment (TheCedarPrince) [inline src/preprocessing.jl:275]: Yup, I think this is correct too! Well done Jay!

**Review by TheCedarPrince — COMMENTED** [review]
Hey @Jay-sanjay , I just wanted to say I think you implemented the time logic here correctly! What else do you need to do to get this PR ready for merging?

**Comment by Jay-sanjay** [comment]
@TheCedarPrince  That's great, then I can add some test cases, for it. And if things go well, we can merge this !

**Comment by TheCedarPrince** [comment]
Sounds great! Keep up the good work!

**Comment by TheCedarPrince** [comment]
Hey @Jay-sanjay , since we discussed the failure of 1.6 and decided to forego testing on 1.6 for the future, I am going to go ahead and merge this in. Thanks for the good work!

**Pull request merged**